test/spec/spectest.cpp

```diff
@@ -128,83 +160,54 @@ parseValueList(const rapidjson::Value &Args) {
+        Result.emplace_back(
+            static_cast<uint64_t>(std::stoull(std::string(Value))));
         ResultTypes.emplace_back(WasmEdge::ValType::F64);
       } else {
         assumingUnreachable();
       }
-    } else if (ValueNode.IsArray()) {
-      WasmEdge::uint64x2_t I64x2;
-      const auto LaneType = Element["lane_type"].Get<std::string>();
-      if (LaneType == "i64"sv || LaneType == "f64"sv) {
-        for (rapidjson::SizeType I = 0; I < 2; ++I) {
-          I64x2[I] = std::stoull(ValueNode[I].Get<std::string>());
-        }
-      } else if (LaneType == "i32"sv || LaneType == "f32"sv) {
-        using uint32x4_t = uint32_t __attribute__((vector_size(16)));
-        uint32x4_t I32x4 = {0};
-        for (rapidjson::SizeType I = 0; I < 4; ++I) {
-          I32x4[I] = std::stoul(ValueNode[I].Get<std::string>());
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I32x4);
-      } else if (LaneType == "i16"sv) {
-        using uint16x8_t = uint16_t __attribute__((vector_size(16)));
-        uint16x8_t I16x8 = {0};
-        for (rapidjson::SizeType I = 0; I < 8; ++I) {
-          I16x8[I] = static_cast<uint16_t>(
-              std::stoul(ValueNode[I].Get<std::string>()));
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I16x8);
-      } else if (LaneType == "i8"sv) {
-        using uint8x16_t = uint8_t __attribute__((vector_size(16)));
-        uint8x16_t I8x16 = {0};
-        for (rapidjson::SizeType I = 0; I < 16; ++I) {
-          I8x16[I] =
-              static_cast<uint8_t>(std::stoul(ValueNode[I].Get<std::string>()));
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I8x16);
-      }
-      Result.emplace_back(I64x2);
-      ResultTypes.emplace_back(WasmEdge::ValType::V128);
-    } else {
-      assumingUnreachable();
     }
   }
   return {Result, ResultTypes};
 }
 
 // Helper function to parse parameters from json to vector of string pair.
 std::vector<std::pair<std::string, std::string>>
-parseExpectedList(const rapidjson::Value &Args) {
+parseExpectedList(const simdjson::dom::array &Args) {
   std::vector<std::pair<std::string, std::string>> Result;
-  Result.reserve(Args.Size());
-  for (const auto &Element : Args.GetArray()) {
-    const auto &Type = Element["type"].Get<std::string>();
-    const auto &ValueNode = Element["value"];
-    if (ValueNode.IsString()) {
-      Result.emplace_back(Type, ValueNode.Get<std::string>());
-    } else if (ValueNode.IsArray()) {
-      const auto LaneType = Element["lane_type"].Get<std::string>();
-      std::string Value;
-      for (auto Iter = ValueNode.Begin(); Iter != ValueNode.End(); ++Iter) {
-        Value += Iter->Get<std::string>();
-        Value += ' ';
+  Result.reserve(Args.size());
+  simdjson::dom::array ValueNodeArray;
+  std::string_view Value;
+  for (const simdjson::dom::object &Element : Args) {
+    std::string_view Type = Element["type"];
+    if (!Element["value"].get(ValueNodeArray)) {
```

Comment:
Does SIMDJSON doesn't provide any function like `is_array`? Why do we always have to create a temp variable for testing the result?

Reply:
Will look into whether or not is there a better implementation for the same.

Reply:
I am now using the `type()` function in the latest commit